pkg/cmd/server/kubernetes/node/node_config.go

```diff
@@ -237,7 +237,18 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 	internalKubeInformers := kinternalinformers.NewSharedInformerFactory(kubeClient, proxyconfig.ConfigSyncPeriod)
 
 	// Initialize SDN before building kubelet config so it can modify option
-	sdnPlugin, err := sdnplugin.NewNodePlugin(options.NetworkConfig.NetworkPluginName, originClient, kubeClient, internalKubeInformers, options.NodeName, options.NodeIP, options.NetworkConfig.MTU, proxyconfig.KubeProxyConfiguration)
+	sdnNodeConfig := &sdnplugin.OsdnNodeConfig{
+		PluginName:      options.NetworkConfig.NetworkPluginName,
+		Hostname:        options.NodeName,
+		SelfIP:          options.NodeIP,
+		RuntimeEndpoint: options.DockerConfig.DockerShimSocket,
+		MTU:             options.NetworkConfig.MTU,
+		OSClient:        originClient,
+		KClient:         kubeClient,
+		KubeInformers:   internalKubeInformers,
+		ProxyConfig:     proxyconfig.KubeProxyConfiguration,
```

Comment:
I changed the code to pass KubeProxyConfiguration in #14466 to avoid adding an extra arg to NewNodePlugin(). You might want to undo that here, and instead have IPTablesSyncPeriod and ProxyMode members of OsdnNodeConfig.